File: examples/csharp_demo/Orders/Order.cs
using System.ComponentModel;
using CodeConstraints.Rules;

namespace Orders;

public class Order
{
    public Cart Cart { get; set; }
    public OrderStatus Status { get; set; } = OrderStatus.Pending;

    public Order(Cart cart)
    {
        Cart = cart;
    }

    // Method-level [NoInstantiation] (complements the class-level example in
    // Billing.cs). `Place` is pure orchestration — it calls existing
    // collaborators and flips an enum, but never constructs a domain object, so
    // it satisfies the rule. Add a `new Receipt(...)` here and `cdec enforce`
    // would flag it; building a whitelisted type would need `Allow = new[] {...}`.
    [NoInstantiation]
    public OrderStatus Place()
    {
        // <uml-sequence name="order_place" root="this">
        var cart_2 = new Cart();
        var result = Cart.Checkout(Cart.Total());
        if (result == "ok")
        {
            Cart.Customer.Charge(Cart.Total());
            Cart.Customer.Notifier.Send(Cart.Customer.Email, "Order placed");
            Status = OrderStatus.Placed;
        }
        else
        {
            Status = OrderStatus.Cancelled;
        }
        return Status;
        // </uml-sequence>
    }
}

File: examples/python_demo/orders/order.py
"""Order placement — the central sequence diagram of the demo."""

from orders.cart import Cart
from orders.status import OrderStatus


class Order:
    def __init__(self, cart: Cart) -> None:
        self.cart: Cart = cart
        self.status: OrderStatus = OrderStatus.PENDING

    def place(self) -> OrderStatus:
        # <uml-sequence name="order_place" root="self">
        result = self.cart.checkout(self.cart.total())
        if result == "ok":
            self.cart.customer.charge(self.cart.total())
            self.cart.customer.notifier.send(
                self.cart.customer.email, "Order placed"
            )
            self.status = OrderStatus.PLACED
        else:
            self.status = OrderStatus.CANCELLED
        return self.status
        # </uml-sequence>
